rust/src/ostree_diff.rs

```diff
@@ -120,12 +117,10 @@ fn diff_recurse(
                     diff.changed_files.insert(path);
                 }
             }
-        } else {
-            if is_dir {
+        } else if is_dir {
```

Comment:
I think this one was debated before already. I can revert it for now, though if we eventually do want to turn on clippy in CI, it seems like a small price to pay. (Though we could also `#[allow(...)]` it for this block too I suppose.)

Reply:
I'd lean towards the `#[allow]`.